airbyte-db/lib/src/main/java/io/airbyte/db/DataTypeUtils.java

```diff
@@ -40,6 +44,12 @@
     }
   }
 
+  public static String toISO8601StringWithMicroseconds(Instant instant) {
+    var microSeconds = (instant.getNano() / 1000) % 1000;
+    String dateWithMilliseconds = DATE_FORMAT_WITH_MILLISECONDS.format(Date.from(instant));
+    return dateWithMilliseconds.substring(0, 23) + microSeconds + dateWithMilliseconds.substring(23);
```

Comment:
Can this use `DateTimeFormatter`? I believe that class supports microseconds natively, rather than needing this manual handling

Reply:
I tried using `DateTimeFormatter` with microseconds - `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'` but unfortunately it doesn't work correctly.

Reply:
what was the error you got? I was able to run this code:
```java
final DateTimeFormatter f = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'").withZone(ZoneOffset.UTC);
System.out.println(f.format(Instant.now()));
```

is the problem that we don't actually know what timezone to use?

Reply:
Not certainly in that way. For a `DateTimeFormatter` of type `yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'`, a microsecond conversion error will occur. For example if your timestamp is like `2018-05-08T02:07:23.123456Z` - this DateTimeFormatter will convert it to `2018-05-08T02:07:23.000123Z`

Reply:
Yes, you're right, that's really better. I added these changes to the pull request. Thank you.

Reply:
@edgao I ran several tests. all the same, for some connectors, this DateTimeFormatter does not work correctly. this breaks hours and minutes and we get an incorrect result at the output. For example oracle and snowflake sources interpret this as minutes and seconds. Therefore, I will revert my changes.